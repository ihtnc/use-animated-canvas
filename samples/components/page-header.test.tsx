import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import PageHeader from './page-header'

describe('PageHeader component', () => {
  test('should render correctly', () => {
    const { container } = render(<PageHeader />)

    expect(container).toMatchSnapshot()
  })

  test('should render two anchor elements', () => {
    const { container } = render(<PageHeader />)

    const links = container.querySelectorAll('a')

    expect(links).not.toBeNull()
    expect(links!).toHaveLength(2)
  })

  test('should render link for the package', () => {
    const { container } = render(<PageHeader />)

    const links = container.querySelectorAll('a')

    let link: HTMLAnchorElement | null = null
    links.forEach((item) => {
      if (item.text === 'package') {
        link = item
      }
    })

    expect(link).not.toBeNull()
  })

  test('should render link for the repo', () => {
    const { container } = render(<PageHeader />)

    const links = container.querySelectorAll('a')

    let link: HTMLAnchorElement | null = null
    links.forEach((item) => {
      if (item.text === 'repo') {
        link = item
      }
    })

    expect(link).not.toBeNull()
  })
})