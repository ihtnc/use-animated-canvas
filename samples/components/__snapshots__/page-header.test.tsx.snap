// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader component > should render correctly 1`] = `
<div>
  <h1
    class="mb-4 ml-4 mr-4 text-3xl text-center font-semibold"
  >
    use2dAnimatedCanvas Examples
  </h1>
  <span
    class="mb-4 flex row"
  >
    <span
      class="flex hover:gap-x-1.5 hover:ml-0 hover:mr-0 group-hover:gap-x-1.5 group-hover:ml-0 group-hover:mr-0 ml-1.5 mr-1.5 cursor-pointer"
    >
      <span>
        [
      </span>
      <a
        class="text-blue-500 text-nowrap"
        href="https://www.npmjs.com/package/@ihtnc/use-animated-canvas"
      >
        package
      </a>
      <span>
        ]
      </span>
    </span>
    <span
      class="flex hover:gap-x-1.5 hover:ml-0 hover:mr-0 group-hover:gap-x-1.5 group-hover:ml-0 group-hover:mr-0 ml-1.5 mr-1.5 cursor-pointer"
    >
      <span>
        [
      </span>
      <a
        class="text-blue-500 text-nowrap"
        href="https://github.com/ihtnc/use-animated-canvas"
      >
        repo
      </a>
      <span>
        ]
      </span>
    </span>
  </span>
</div>
`;
